lms/static/js/student_profile/views/badge_listing_view.js

```diff
@@ -0,0 +1,37 @@
+;(function (define, undefined) {
+    'use strict';
+    define([
+            'gettext', 'jquery', 'underscore', 'backbone', 'js/student_profile/views/badge_view',
+            'text!templates/student_profile/badge_placeholder.underscore'],
+        function (gettext, $, _, Backbone, BadgeView, badgePlaceholder) {
+
+            var BadgeListingView = Backbone.View.extend({
+                initialize: function (options) {
+                    this.find_courses_url = options.find_courses_url;
+                },
+                render: function () {
+                    var grid = $('<div class="badge-set-display">');
+                    var row;
+                    this.collection.each(function (badge, index, collection) {
+                        var keep_row = (index % 2);
+                        if (!keep_row) {
+                            row = $("<div class='badge-row'>");
```

Comment:
I just tested and the HTML appears the same to me whether you put all columns ( > 2 ) inside a single "row" element or whether you manually split them up into rows. I'd advise just using one big "row" element and put all the badges into that single row, which is more flexible and simpler compared to manually assigning them into rows. Maybe just rename the class so it's not called a "row".